dvc/repo/reproduce.py

```diff
@@ -249,7 +250,7 @@ def reproduce(
         logger.debug("Pulling run cache")
         try:
             self.stage_cache.pull(None)
-        except RunCacheNotSupported as e:
+        except (RunCacheNotSupported, RemoteConfigError) as e:
```

Comment:
We should probably catch RemoteConfigError in stagecache.pull and re-raise it as RunCacheNotSupported, which is already used for http and seems to fit the logic for cloud-versioned remotes too.

Reply:
PTAL 🙏